### PR TITLE
[5.2] Catch exception to get the user in the action log model

### PR DESCRIPTION
**Pull request by laoneo**
Pull Request for Issue #42373.

### Summary of Changes
Catches an exception when the user factory is not set in the action log model. This is basically against the principle that we want to have the dependencies injected, but it is good for the transition period till all extensions are up to the new architecture.

### Testing Instructions
Happens only on 3rd party extensions, see issue.

### Actual result BEFORE applying this Pull Request
Extension produces an error when writing to the action log.

### Expected result AFTER applying this Pull Request
Extension works and action log is written to the database.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_actionlogs/src/Model/ActionlogModel.php [hunk 1]
```diff
@@ -55,7 +55,13 @@ public function addLog($messages, $messageLanguageKey, $context, $userId = 0)
             @trigger_error(\sprintf('User ID must be an integer in %s.', __METHOD__), E_USER_DEPRECATED);
         }
 
-        $user   = $userId ? $this->getUserFactory()->loadUserById($userId) : $this->getCurrentUser();
+        try {
+            $user   = $userId ? $this->getUserFactory()->loadUserById($userId) : $this->getCurrentUser();
```
Comment (Quy) [inline administrator/components/com_actionlogs/src/Model/ActionlogModel.php:59]:
```suggestion
            $user = $userId ? $this->getUserFactory()->loadUserById($userId) : $this->getCurrentUser();
```

Comment (Quy) [inline administrator/components/com_actionlogs/src/Model/ActionlogModel.php]:
```suggestion
            $user = $userId ? $this->getUserFactory()->loadUserById($userId) : $this->getCurrentUser();
```

**Comment by ssnobben** [comment]
Yes the solution for  seems working ok now after this fix the error message gone. Great! This still an issue wihth the Finder plg that have to be disable to though..    Plugin\Finder\Content\Extension\Content::onFinderAfterSave() error #42617 https://github.com/joomla/joomla-cms/issues/42617

**Comment by laoneo** [comment]
The finder issue is not addressed by this pr.

**Comment by ssnobben** [comment]
> The finder issue is not addressed by this pr.

Yes ok. Thks Allon .

**Comment by ssnobben** [comment]
Can anyone test this one "by the boook" Joomla way so its into next release? or by code review bcs Its working now after this fix. Thks.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44358">issues.joomla.org/tracker/joomla-cms/44358</a>.</sub>

**Comment by Hackwar** [comment]
@ssnobben if you were to mark this as successfully tested, then we could merge this into 5.2.3.

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on e4a6c79c4ac25d86b3ffb4acc22daeab808e3ae6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44358">issues.joomla.org/tracker/joomla-cms/44358</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44358">issues.joomla.org/tracker/joomla-cms/44358</a>.</sub>

**Comment by Hackwar** [comment]
Every PR for the CMS needs 2 successful manual tests. At the time of my comment, I was test 1 and you could have been test 2. Now @viocassel has thankfully tested it. Everything else is not of concern right now.

**Comment by ssnobben** [comment]
Thanks @viocassel  !

**Comment by ssnobben** [comment]
Ok so why was this important issue not merged?? I want to hear an explanation.. I have beew working with Joomla and 3pds that is important for Joomla eccosystem since 2005 and this have been the most waited fix ever...

**Comment by pe7er** [comment]
> Ok so why was this important issue not merged?? I want to hear an explanation.. I have beew working with Joomla and 3pds that is important for Joomla eccosystem since 2005 and this have been the most waited fix ever...

The second human test for this PR came in too late to include it in Joomla 5.2.2, 
which was already scheduled and released earlier today, just 2.5 hours ago.

**Comment by Hackwar** [comment]
@ssnobben Your comment is completely out of line.

**Comment by ssnobben** [comment]
> @ssnobben Your comment is completely out of line.

Could be but this have been an issue since 2023 until it finally got an solution. - Thu Oct 26, 2023 10:22 am reported here in Joomla world first. I work with 3pds and people have waiting for a solution for this to release their site forever...

**Comment by pe7er** [comment]
> Could be but this have been an issue since 2023 until it finally got an solution. - Thu Oct 26, 2023 10:22 am reported here in Joomla world first. I work with 3pds and people have waiting for a solution for this to release their site forever...

I understand your concern. However, every change can affect other parts of Joomla or extensions from third parties. Therefore we need to be careful with adding any code to Joomla. Every PR needs to pass one successful automated test and get two successful human tests before it can be added to the core. One week before a release, a Release Candidate is created for testing that release. At this stage, we do not add new PRs, but only fix issues with the existing code in that package. The second successful test for this PR was reported during this phase so it was too late to add it to 5.2.2.

**Comment by Hackwar** [comment]
We have issues reported as old as 2015 and serious issues for example with MFA from 2020. As unfortunate as it is that it took a year to get it ready, it is not uncommon. And the reason is very simple: We are volunteers and do this in our spare time. You were very welcome to hand in a patch/PR yourself to fix this in all that time. You are not required to first open an issue. If you find a problem, you can directly create a PR and if you then motivate a few other people to test the PR, it can actually merged pretty quickly. If you don't feel that you have the knowledge to do that, you can also hire a developer to do this for you. We don't really discriminate based on this. But if you are unwilling to either work on this yourself or to invest into this and instead want free work from the community, you have to accept that it might sometimes take years until an issue is solved.

**Comment by pe7er** [comment]
Thanks @laoneo !

**Pull request merged**